job.go

```diff
@@ -55,22 +53,13 @@ func JobCompleteTx[TDriver riverdriver.Driver[TTx], TTx any, TArgs JobArgs](ctx
 		return nil, errors.New("job must be running")
 	}
 
-	var (
-		driver  TDriver
-		queries = &dbsqlc.Queries{}
-	)
-
-	internal, err := queries.JobSetState(ctx, driver.UnwrapTx(tx), dbsqlc.JobSetStateParams{
-		ID:                  job.ID,
-		FinalizedAtDoUpdate: true,
-		FinalizedAt:         ptrutil.Ptr(time.Now()),
-		State:               dbsqlc.JobStateCompleted,
-	})
+	var driver TDriver
+	jobRow, err := driver.UnwrapExecutor(tx).JobSetStateIfRunning(ctx, riverdriver.JobSetStateCompleted(job.ID, time.Now()))
```

Comment:
This is a small behavioral change due to the differences in the underlying queries, but I don't think it's a big issue. You really shouldn't be using this to set a job as complete if it's not actually running.

Reply:
Yeah true, but I think you're right in that it won't matter. If being called in a `Work` function then the job should be running unless something weird is going on. Feels good to get rid of a big SQL blob.